contracts/market/Market.sol

```diff
@@ -104,7 +108,17 @@ contract Market is UsingConfig {
 
 	function authenticatedCallback(address _prop) public returns (address) {
 		Metrics metrics = new Metrics(_prop);
-		MetricsGroup(config().metricsGroup()).addMetrics(address(metrics));
+		MetricsGroup metricsGroup = MetricsGroup(config().metricsGroup());
+		metricsGroup.addMetrics(address(metrics));
+		uint256 tokenValue = Lockup(config().lockup()).getTokenValueByProperty(
+			metrics.property()
+		);
+		Policy policy = Policy(config().policy());
+		uint256 authenticationFee = policy.authenticationFee(
+			metricsGroup.totalIssuedMetrics(),
+			tokenValue
+		);
```

Comment:
Here, the second args should indicate **the number of Metrics** related to **the Property**.

Because so, I think MetricsGroup requires a method that returns the number of Metrics per Property.

Reply:
Sorry, wait, this code may be no problem. It seems that this code can meet the Policy specification. No commit is required.

Reply:
This fee work to brakes the increase in the number of assets. The total number of assets is more important than the number of assets for each Property because the number of Properties is unlimited. And it's already in your code.

Also, if the lock-up proportional to the increase in assets, it will have a positive impact on the DEV price. So, a property with more lock-ups can reduce fees.

In other words, it is better to rewrite the whitepaper to match this code than to change this code.